From d0fa719d49f38777911a8e99717f2d4b30811080 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 12 Feb 2024 19:24:54 -0800
Subject: [PATCH] cmake: Test for sys/uio.h

writev() API is used in few places and it will need including sys/uio.h
for getting the funciton signature

Upstream-Status: Submitted [https://github.com/sahlberg/libnfs/pull/440]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 cmake/ConfigureChecks.cmake | 1 +
 cmake/config.h.cmake        | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/cmake/ConfigureChecks.cmake b/cmake/ConfigureChecks.cmake
index 0258fc8..c26a02d 100644
--- a/cmake/ConfigureChecks.cmake
+++ b/cmake/ConfigureChecks.cmake
@@ -22,6 +22,7 @@ check_include_file("sys/stat.h" HAVE_SYS_STAT_H)
 check_include_file("sys/sysmacros.h" HAVE_SYS_SYSMACROS_H)
 check_include_file("sys/time.h" HAVE_SYS_TIME_H)
 check_include_file("sys/types.h" HAVE_SYS_TYPES_H)
+check_include_file("sys/uio.h" HAVE_SYS_UIO_H)
 check_include_file("sys/vfs.h" HAVE_SYS_VFS_H)
 check_include_file("unistd.h" HAVE_UNISTD_H)
 check_include_file("utime.h" HAVE_UTIME_H)
diff --git a/cmake/config.h.cmake b/cmake/config.h.cmake
index 4016c6c..6263896 100644
--- a/cmake/config.h.cmake
+++ b/cmake/config.h.cmake
@@ -90,6 +90,9 @@
 /* Define to 1 if you have the <sys/types.h> header file. */
 #cmakedefine HAVE_SYS_TYPES_H
 
+/* Define to 1 if you have the <sys/uio.h> header file. */
+#cmakedefine HAVE_SYS_UIO_H
+
 /* Define to 1 if you have the <sys/vfs.h> header file. */
 #cmakedefine HAVE_SYS_VFS_H
 
-- 
2.43.1

